routes/artifacts.js

```diff
@@ -402,6 +403,11 @@ var replyWithArtifact = async function(taskId, runId, name, req, res) {
           // are in the canonical region
           url = canonicalArtifactUrl;
         } else {
+          // If the region is not resolved to anything, we assume that it's in the canonical
+          // region.  This should go away when we move region resolving into cloud-mirror
+          if (!region) {
+            region = this.artifactRegion;
```

Comment:
Should you make the condition above:
`if (!region || this.artifactRegion === region) {` ?


hmm... that's also wrong too, as if region isn't resolved we should return the cloudfront url...

If we don't have a region shouldn't we be using `this.publicBucket.createGetUrl(prefix);` as that is the url to the CDN...